plugins/meta/flannel/flannel.go

```diff
@@ -46,6 +46,8 @@ const (
 type NetConf struct {
 	types.NetConf
 
+	// IPAM field "replaces" that of types.NetConf which is incomplete
+	IPAM          map[string]interface{} `json:"ipam,omitempty"`
```

Comment:
Why use a string map rather than a struct ?
(It came as a surprise to me that `containernetworking/cni` does not return any useful IPAM struct,  but even so, why don't you?)

Reply:
Even though my own use case is about the routes, I thought it would be nice to allow transparently passing any IPAM parameter that could apply to the delegate (bridge or other). In order not to limit the possibilities there, a string map looked appropriate to me (I'm not aware of any way to unmarshal into a struct while preserving and giving access to all extra fields not defined in the struct).
You will see in flannel_linux.go:getDelegateIPAM() that the incoming string map, when present, is used as basis for the delegate ipam.